### DISCUSSION TITLE
labelling a lepton jet that specifically decays from top and anti-top.

### DISCUSSION THREAD

**Discussion by randomizer6542** [MadAnalysis/madanalysis5#192] [Q&A]
Hi,
This might be a simple question but I am stuck on this, I am learning madanalysis and I want to specify leptons that have decayed from a top/anti-top quark under a label l3
something like
l3 = electrons and muons decay from top and anti-top decay into W and b
can someone help me out with this?

Regards,

**Comment by BFuks** [comment]
Hi,

That’s easy to do as you can enforce the nature of the mother of the particle considered. For instance, using the normal mode of the code, you can type
 define tt = 6 -6
 define ll = 11 13 -13 -11
 plot PT(ll << tt)
This should do the required job.

Cheers,

Benjamin

PS: this information is available from the code’s manual, page 52...

**Reply by randomizer6542** [reply]
Just a clarification, in the analysis, I am using 4-5 processes, with leptons in them already, (though 3 of them not decayed directly through t > l+ vl b) and I have defined two general labels for the leptons already as l1 = l+ and l2 = l- so will this process as you defined be able to retain it's identity? I just want the code to confuse this with others as I will be plotting them all 3 separately (Transverse masses of l1 and l2 together and l3 with b and some other factors)

Also can you link the manual, that will help a lot

**Reply by jackaraz** [reply]
Hi @randomizer6542, as long as you are not using reconstructed events madanalysis is capable of looking into the event history and extract the decay information of all your particles. That being said this depends if top is on-shell because if its off-shell it is not written in your event file so Ma5 wont be able to extract that information. It is important to note that Ma5 only looks for the first parent of the particle so since t > b W > l nu if W is onshell Ma5 wont be able to identify that lepton as top’s product in normal mode. Hence for such decay chains you will need to use expert-mode where you can access all the information you need through our C++ backend.

**Reply by jackaraz** [reply]
**Sorry correction** It actually looks for the entire decay chain so it doesnt matter how many decays you have if there is a top in your leptons decay chain Ma5 will be able to find it.

**Reply by randomizer6542** [reply]
@BFuks @jackaraz appreciate the response from both, I will look it up.

**Reply by randomizer6542** [reply]
> **Sorry correction** It actually looks for the entire decay chain so it doesnt matter how many decays you have if there is a top in your leptons decay chain Ma5 will be able to find it.

@jackaraz Is there anyway to do this if i am using the reco mode or is it only aplicable when I am using normal mode. i.e. assigning the labels to particles and then plotting for example Plot M(l1<<t) ?

**Comment by BFuks** [comment]
Hi,

Here is the link to the manual:
  https://inspirehep.net/literature/1117708
Note that it is quite old, and you may need to check out more recent papers for updates. See the list at the end of this page:
  https://github.com/MadAnalysis/madanalysis5

Cheers,

Benjamin

**Comment by BFuks** [comment]
Hi,

This information is lost in the reconstruction process, so that the answer is “no”.

Cheers,

Benjamin
